### PR TITLE
OHRM5X-2501

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-2501: Fix Stored XSS via Client Logo
- OHRM5X-2513: [CLI Installer][PHP82][PHP83] Displays an error message when the Password Field contains a value that doesn't meet password policy requirements
- OHRM5X-2514: [CLI Installer][Password Enforcement] CLI installer doesn't allow user to proceed the installation process with weak password
- OHRM5X-2502: Update node-ip package once it's patched
- OHRM5X-2507: Update phpseclib package
- OHRM5X-2517: Update follow-redirects package

### CODE REVIEW

File: src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php [hunk 1]
```diff
@@ -228,12 +230,29 @@ public function getAttachmentOrNull(string $type, string $key, ?array $default =
...
-            return new Base64Attachment(
+            $attachment = new Base64Attachment(
                 $attachment['name'],
                 $attachment['type'],
                 $attachment['base64'],
                 $attachment['size']
             );
+
+            // Check for SVG and sanitize
+            if ($attachment->getFileType() === 'image/svg+xml') {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php:241]: have a look on these mime types also `text/xml`, `application/svg+xml`, `image/svg`

File: src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php [hunk 2]
```diff
@@ -228,12 +230,29 @@ public function getAttachmentOrNull(string $type, string $key, ?array $default =
...
+                $sanitizer = new Sanitizer();
+                try {
+                    $sanitizedContent = $sanitizer->sanitizeSvg($attachment->getContent());
+
+                    $attachment->setContent($sanitizedContent);
+                    $attachment->setSize(strlen($sanitizedContent));
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php:247]:
```suggestion
                    $attachment->setSize($this->getTextHelper()->strLength($sanitizedContent, '8bit'));
```
Check;
`src/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/Base64Attachment.php`
inside the helper we used `mb_strlen` to handle unicode. have a look

**Review by RajithaKumara — APPROVED** [review]

File: src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php (hunk 2 above)
Comment (devishke-orange) [reply src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php]: changed to use `TextHelper`

File: src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php (hunk 1 above)
Comment (devishke-orange) [reply src/plugins/orangehrmCorePlugin/Api/V2/RequestParams.php:244]: seems image/svg+xml is the main one. our API also only allows for that

https://www.w3.org/TR/SVG11/intro.html#MIMEType
https://datatracker.ietf.org/doc/html/rfc3023#section-8.19
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

**Pull request merged**